Sources/LanguageServerProtocol/SupportTypes/ServerCapabilities.swift

```diff
@@ -773,21 +839,45 @@ public struct DocumentLinkOptions: WorkDoneProgressOptions, Codable, Hashable {
   }
 }
 
-public struct SemanticTokensOptions: WorkDoneProgressOptions, Codable, Hashable {
+public struct SemanticTokensOptions: WorkDoneProgressOptions, Codable, Hashable, LSPAnyCodable {
 
-  public struct SemanticTokensRangeOptions: Equatable, Hashable, Codable {
+  public struct SemanticTokensRangeOptions: Equatable, Hashable, Codable, LSPAnyCodable {
     public init() {
       // Empty in the LSP 3.16 spec.
     }
+
+    public init?(fromLSPDictionary dictionary: [String : LSPAny]) {
+      nil
+    }
+
+    public func encodeToLSPAny() -> LSPAny {
+      .dictionary([:])
+    }
   }
 
-  public struct SemanticTokensFullOptions: Equatable, Hashable, Codable {
+  public struct SemanticTokensFullOptions: Equatable, Hashable, Codable, LSPAnyCodable {
     /// The server supports deltas for full documents.
     public var delta: Bool?
 
     public init(delta: Bool? = nil) {
       self.delta = delta
     }
+
+    public init?(fromLSPDictionary dictionary: [String : LSPAny]) {
+      self.delta = nil
+
+      if case .bool(let value) = dictionary["delta"] {
+        self.delta = value
+      }
+    }
+
+    public func encodeToLSPAny() -> LSPAny {
+      guard let delta else {
+        return .null
```

Comment:
Shouldn’t we return an empty dictionary here instead of `null`? Mostly so that encoding + decoding to `LSPAny` round-trips.